src/registers/model_specific.rs

```diff
@@ -725,4 +753,69 @@ mod x86_64 {
             }
         }
     }
+
+    impl Apic {
+        /// Reads the IA32_APIC_BASE.
+        ///
+        /// The APIC_BASE must be supported on the CPU, otherwise a general protection exception will
+        /// occur. Support can be detected using the `cpuid` instruction.
```

Comment:
```suggestion
        /// IA32_APIC_BASE must be supported on the CPU, otherwise, a general protection exception will
        /// occur. Support can be detected using the `cpuid` instruction.
```

Having this comment on every method seems a bit repetitive. How about we move it to the struct's definition?